gateway/src/node/client_handling/websocket/connection_handler/fresh.rs

```diff
@@ -802,9 +694,6 @@ impl<R, S> FreshHandler<R, S> {
             .await?;
         let shared_keys = handshake_result.derived_key;
 
-        // populate the negotiated protocol for future uses
-        self.negotiated_protocol = Some(handshake_result.negotiated_protocol);
```

Comment:
well no. you should take the result from the handshake because that's where the negotiation occurs.  the passed value is just an initial proposal